comit/src/expiries.rs

```diff
@@ -523,119 +691,81 @@ impl From<BobState> for AliceState {
+        (MockConnector, MockConnector)
     }
 
-    /// The duration of time it takes for Bob to act.
-    pub fn bob_to_act(&self) -> Duration {
-        1.hours()
-    }
+    #[test]
+    fn can_calculate_useful_expiries_for_all_supported_protocols() {
+        let scale = None;
+        let future = Timestamp::now().plus(10);
+        let (alpha, beta) = mock_connectors();
 
-    /// The duration of time it takes for the alpha fund transaction to be
-    /// mined into the blockchain.
-    pub fn mine_alpha_fund_transaction(&self) -> Duration {
-        1.hours()
-    }
+        let exp = Expiries::new(Protocol::Herc20Hbit, alpha, beta, scale);
+        assert!(exp.is_useful(future));
 
-    /// The duration of time it takes for the beta fund transaction to be
-    /// mined into the blockchain.
-    pub fn mine_beta_fund_transaction(&self) -> Duration {
-        1.hours()
+        let exp = Expiries::new(Protocol::HbitHerc20, alpha, beta, scale);
+        assert!(exp.is_useful(future));
     }
 
-    /// The duration of time it takes for the alpha redeem transaction to be
-    /// mined into the blockchain.
-    pub fn mine_alpha_redeem_transaction(&self) -> Duration {
-        1.hours()
-    }
+    #[test]
+    fn can_calculate_useful_expiries_for_all_supported_protocols_with_scale() {
+        let scale = Some(120);
+        let now = Timestamp::now();
+        let (alpha, beta) = mock_connectors();
 
-    /// The duration of time it takes for the beta redeem transaction to be
-    /// mined into the blockchain.
-    pub fn mine_beta_redeem_transaction(&self) -> Duration {
-        1.hours()
-    }
+        let exp = Expiries::new(Protocol::Herc20Hbit, alpha, beta, scale);
+        assert!(exp.is_useful(now));
 
-    /// The duration of time it takes for a transaction to reach finality on the
-    /// alpha ledger.
-    pub fn finality_alpha(&self) -> Duration {
-        1.hours()
+        let exp = Expiries::new(Protocol::HbitHerc20, alpha, beta, scale);
+        assert!(exp.is_useful(now));
     }
 
-    /// The duration of time it takes for a transaction to reach finality on the
-    /// beta ledger.
-    pub fn finality_beta(&self) -> Duration {
-        1.hours()
+    #[test]
+    fn scale_by_factor_100_works() {
+        let d = Duration::minute();
+        let scaled = scale_by_factor(d, 100);
+        assert_eq!(d, scaled)
     }
-}
-
-fn bitcoin_required_confirmations() -> usize {
-    // TODO: Add documentation on _why_ we picked this value.
-    6
-}
-
-fn ethereum_required_confirmations() -> usize {
-    // TODO: Add documentation on _why_ we picked this value.
-    12
-}
-
-#[cfg(test)]
-mod tests {
-    use super::*;
 
-    #[derive(Clone, Copy, Debug)]
-    struct MockConnector;
-
-    impl CurrentTime for MockConnector {
-        fn current_time(&self) -> Timestamp {
-            Timestamp::now()
-        }
-    }
+    #[test]
+    fn scale_by_factor_200_works() {
```

Comment:
what about `0` and under `100`?

Reply:
Sure thing, under 100 catches 0, I'll add it.

Reply:
Re-named the function to `scale_up_by_factor` and added check for factor < 100. Added unit test to test this functionality.

Reply:
In case its not obvious, I'm using 100 as no scale so we get finer control without using floats.